### PR TITLE
Remove hard-coded English strings from MyData API

### LINKED ISSUE #9292: Translate hard coded text - My Data page
**What steps does it take to reproduce the issue?**
When going to My Data menu on a user without any permissions (just registered).

* When does this issue occur?
Always

* Which page(s) does it occurs on?
/dataverseuser.xhtml?selectTab=dataRelatedToMe

* What happens?
No possible translation of alert text: (Sorry, nothing was found for these roles:)

* To whom does it occur (all users, curators, superusers)?
registered users

* What did you expect to happen?
make strings available to translate that message

**Which version of Dataverse are you using?**
5.12.1

**Any related open or closed issues to this bug report?**
Didn't find any

**Screenshots:**
Screenshot from 2023-01-17 12-02-40

### PR DESCRIPTION
**Pull request by stevenferey**
**What this PR does / why we need it**:

**Which issue(s) this PR closes**:

Closes #9292 

**Special notes for your reviewer**:

**Suggestions on how to test this**:

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

**Review by qqmyers — APPROVED** [review]
Looks good - did not test but the auto-test and QA should be able to verify the messages aren't getting changed. 
FWIW: There are a couple weird text strings in the original that this PR faithfully makes internationalizable, i.e. "retrieveMyDataAsJsonString. User not found! Shouldn't be using this anyway.", "Sorry, the EntityManager isn't working (still).". If anyone knows more about when they might appear, perhaps some more user-oriented text could be substituted.

File: src/main/java/propertyFiles/Bundle.properties [hunk 1]
```diff
@@ -2794,7 +2794,22 @@ passwdVal.passwdReq.lowercase=lowercase
 passwdVal.passwdReq.letter=letter
 passwdVal.passwdReq.numeral=numeral
 passwdVal.passwdReq.special=special
+#mydata API (DataRetriverAPI.java and MyDataFinder.java)
 dataretrieverAPI.noMsgResultsFound=Sorry, no results were found.
+dataretrieverAPI.authentication.required=Requires authentication. Please login.
+dataretrieverAPI.authentication.required.opt=retrieveMyDataAsJsonString. User not found! Shouldn't be using this anyway.
+dataretrieverAPI.user.not.found=No user found for: "{0}"
+dataretrieverAPI.solr.error=Sorry! There was an error with the search service.
+dataretrieverAPI.solr.error.opt=Sorry! There was a SOLR Error.
```
Comment (pdurbin) [inline src/main/java/propertyFiles/Bundle.properties:2803]:
```suggestion
dataretrieverAPI.solr.error.opt=Sorry! There was a Solr Error.
```

File: src/main/java/propertyFiles/Bundle.properties [hunk 2]
```diff
@@ -2794,7 +2794,22 @@ passwdVal.passwdReq.lowercase=lowercase
...
+myDataFilterParams.error.no.user=Sorry! No user was found!
+myDataFilterParams.error.result.no.role=No results. Please select at least one Role.
+myDataFilterParams.error.result.no.dvobject=No results. Please select one of Dataverses, Datasets, Files.
+myDataFilterParams.error.result.no.publicationStatus=No results. Please select one of {0}.
+myDataFinder.error.result.null=Sorry, the EntityManager isn't working (still).
```
[the remark this answers is not in the document]
Comment (stevenferey) [reply src/main/java/propertyFiles/Bundle.properties]: A little more contextualization in this case with: "Sorry, the authenticated user ID could not be retrieved."

**Review by qqmyers — APPROVED** [review]
@stevenferey - looks good - probably needs a merge with develop to get the tests to run again.

File: src/main/java/propertyFiles/Bundle.properties [hunk 3]
```diff
@@ -2839,7 +2839,22 @@ passwdVal.passwdReq.lowercase=lowercase
 passwdVal.passwdReq.letter=letter
 passwdVal.passwdReq.numeral=numeral
 passwdVal.passwdReq.special=special
+#mydata API (DataRetriverAPI.java and MyDataFinder.java)
 dataretrieverAPI.noMsgResultsFound=Sorry, no results were found.
+dataretrieverAPI.authentication.required=Requires authentication. Please login.
+dataretrieverAPI.authentication.required.opt=retrieveMyDataAsJsonString. User not found! Shouldn't be using this anyway.
```
Comment (qqmyers) [inline src/main/java/propertyFiles/Bundle.properties:2845]: FWIW: This is still odd, but only shows in the log.

**Comment by sekmiller** [comment]
@stevenferey This PR has a merge conflict. Please resolve. Also I'm seeing a failing test in DataRetrieverApiIT. Can you please take a look at that? Thanks!

**Comment by sekmiller** [comment]
Looks good @stevenferey @jeromeroucou 

Thanks for the contribution and your patience through the process

**Pull request merged**